### PR TITLE
Updates for packaging and release

### PR DESCRIPTION
**Pull request by zsweigart**
Updated the file structure, tsconfig, and package.json to enable publishing based on [this article](https://medium.com/@rossbulat/how-to-develop-typescript-npm-packages-for-react-native-a00f88f8a96c).

All types are now exported out of `index` so they can be imported directly from `'react-native-plaid-link-sdk'`

To test this you need to run `tsc` then copy all the code (except `/node_modules` and `/example`) into `/example/node_modules/react-native-plaid-link-sdk`

### CODE REVIEW

**Review by amytang0 — COMMENTED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -2,8 +2,11 @@
   "name": "react-native-plaid-link-sdk",
   "version": "6.0.3",
   "description": "React Native Plaid Link SDK",
-  "main": "index.js",
-  "types": "/types/Types.d.ts",
+  "main": "lib/index.js",
```
Comment (JeroenMols) [inline package.json:5]: Awesome, so now Javascript is created from typescript! 🔥

File: tsconfig.json [hunk 2]
```diff
@@ -4,17 +4,21 @@
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
     "isolatedModules": true,
-    "jsx": "react",
+    "jsx": "react-native",
```
Comment (JeroenMols) [inline tsconfig.json:7]: should we also change this in the example?

File: tsconfig.json [hunk 3]
```diff
@@ -4,17 +4,21 @@
...
+    "declaration": true,
+    "outDir": "./build",
   },
   "exclude": [
     "node_modules",
     "babel.config.js",
     "metro.config.js",
-    "jest.config.js"
+    "jest.config.js",
+    "example"
```
Comment (JeroenMols) [inline tsconfig.json:22]: 🔥

**Review by JeroenMols — APPROVED** [review]

File: tsconfig.json (hunk 2 above)
Comment (zsweigart) [reply tsconfig.json:7]: Yep, will follow-up

**Review by JeroenMols — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -2,8 +2,11 @@
   "name": "react-native-plaid-link-sdk",
   "version": "6.0.3",
   "description": "React Native Plaid Link SDK",
-  "main": "index.js",
-  "types": "/types/Types.d.ts",
+  "main": "build/index.js",
+  "types": "build/Types.d.ts",
+  "files": [
+    "build/**/*"
```
Comment (zsweigart) [inline package.json:8]: @skylarmb recommended moving to a build folder for compiled code

**Review by JeroenMols — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
-  "main": "index.js",
-  "types": "/types/Types.d.ts",
+  "main": "build/index.js",
+  "types": "build/index.d.ts",
+  "files": [
+    "build/**/*",
+    "android/**/*",
+    "ios/**/*",
+    "react-native-plaid-link-sdk.podspec"
+  ],
```
Comment (skylarmb) [inline package.json:12]: i think a more widely used name for this type of folder would be `dist` instead of `build`.

**Review by skylarmb — DISMISSED** [review]

File: index.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    LinkEventListener,
+    LinkEvent,
+    LinkEventMetadata,
+    LinkEventName,
+    LinkEventViewName,
+    LinkSuccessListener,
+    LinkExitListener,
+    PlaidLinkConfiguration,
+    PlaidLinkProps,
+} from './Types';
```
[the remark this answers is not in the document]
Comment (skylarmb) [reply index.tsx:37]: does this need to be a `.tsx` file? can it just be `.ts`?

**Review by amytang0 — DISMISSED** [review]
LGTM @skylarmb would love your expert eyes on this one last time

File: package.json [hunk 7]
```diff
@@ -1,15 +1,22 @@
...
   "scripts": {
     "lint": "eslint \"./**/*.{js,jsx}\" --fix",
     "prettier": "prettier \"./**/*.{js,jsx}\" --write",
     "test": "echo \"Error: no test specified\" && exit 1",
     "version": "git add -A",
-    "postversion": "git push && git push --tags && rm -rf build/temp"
+    "postversion": "git push && git push --tags && rm -rf build/temp",
```
Comment (skylarmb) [inline package.json:18]: should this be updated to use `dist` as well?

File: tsconfig.json [hunk 8]
```diff
@@ -4,17 +4,21 @@
...
+    "skipLibCheck": true,
     "lib": ["es6"],
     "moduleResolution": "node",
-    "noEmit": true,
     "strict": true,
-    "target": "esnext"
+    "target": "es6",
+    "module": "es6",
+    "declaration": true,
+    "outDir": "./build",
```
Comment (skylarmb) [inline tsconfig.json:15]: dist?

**Review by skylarmb — APPROVED** [review]
lgtm other than `dist` vs `build` consistency, nice work!

**Review by skylarmb — APPROVED** [review]

**Pull request merged**